### PR TITLE
Increase testing coverage on TextArea component

### PR DESCRIPTION
**Pull request by JoeButler7**
#### What does this PR do?
Increases testing coverage for `TextArea` by adding behavioral tests for event handlers. 
Coverage numbers before and after the tests were added are in the following table:
| Type       | Before | After  |
|------------|--------|--------|
| Statements | 73.81% | 95.24% |
| Branches   | 66.67% | 81.82% |
| Functions  | 71.43% | 92.86% |
| Lines      | 79.49% | 94.87% |
#### Where should the reviewer start?
`src/js/components/TextArea/__tests__/TextArea-test.js`
#### What testing has been done on this PR?
ran `yarn test TextArea-test` to ensure that coverage was increased and all the tests passed
#### How should this be manually tested?
See above
#### Screenshots (if appropriate)
Coverage before the tests were updated
Screen Shot 2020-07-01 at 5 14 58 PMCoverage after
Screen Shot 2020-07-01 at 5 13 25 PM

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/TextArea/__tests__/TextArea-test.js [hunk 1]
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
...
       expect(capturedEvent).toEqual(expect.objectContaining(keyEvent));
     });
+
+    test('onFocus', () => {
+      const onFocus = jest.fn();
+      const { container, getByPlaceholderText } = render(
+        <Grommet>
+          <TextArea
+            id="item"
+            name="item"
```
Comment (jcfilben) [inline src/js/components/TextArea/__tests__/TextArea-test.js:158]: Do you need to set the id and name for this test?

File: src/js/components/TextArea/__tests__/TextArea-test.js [hunk 2]
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
...
+            placeholder="item"
+            onFocus={onFocus}
+          />
+        </Grommet>,
+      );
+      fireEvent.focus(getByPlaceholderText('item'));
+      expect(container.firstChild).toMatchSnapshot();
+      expect(onFocus).toHaveBeenCalled();
```
Comment (jcfilben) [inline src/js/components/TextArea/__tests__/TextArea-test.js:166]:
```suggestion
      expect(onFocus).toHaveBeenCalledTimes(1);
```

File: src/js/components/TextArea/__tests__/TextArea-test.js [hunk 3]
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
...
+    });
+
+    test('onChange', () => {
+      const onChange = jest.fn();
+      const { container, getByPlaceholderText } = render(
+        <Grommet>
+          <TextArea
+            id="item"
+            name="item"
```
Comment (jcfilben) [inline src/js/components/TextArea/__tests__/TextArea-test.js:175]: Same thing here with id and name, do they need to be set for the test

File: src/js/components/TextArea/__tests__/TextArea-test.js [hunk 4]
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
...
+            placeholder="item"
+            onChange={onChange}
+          />
+        </Grommet>,
+      );
+      fireEvent.change(getByPlaceholderText('item'), {
+        target: { value: 'o' },
+      });
+      expect(container.firstChild).toMatchSnapshot();
+      expect(onChange).toHaveBeenCalled();
```
Comment (jcfilben) [inline src/js/components/TextArea/__tests__/TextArea-test.js:185]:
```suggestion
      expect(onChange).toHaveBeenCalledTimes(1);
```

File: src/js/components/TextArea/__tests__/TextArea-test.js [hunk 5]
```diff
@@ -148,5 +148,53 @@ describe('TextArea', () => {
...
+    test('onBlur', () => {
+      const onBlur = jest.fn();
+      const { container, getByPlaceholderText } = render(
+        <Grommet>
+          <TextArea id="item" name="item" placeholder="item" onBlur={onBlur} />
+        </Grommet>,
+      );
+      fireEvent.blur(getByPlaceholderText('item'));
+      expect(container.firstChild).toMatchSnapshot();
+      expect(onBlur).toHaveBeenCalled();
```
Comment (jcfilben) [inline src/js/components/TextArea/__tests__/TextArea-test.js:197]:
```suggestion
      expect(onBlur).toHaveBeenCalledTimes(1);
```

File: src/js/components/TextArea/__tests__/TextArea-test.js [hunk 6]
```diff
@@ -163,5 +163,43 @@ describe('TextArea', () => {
...
+
+    test('onChange', () => {
+      const onChange = jest.fn();
+      const { container, getByPlaceholderText } = render(
+        <Grommet>
+          <TextArea name="item" placeholder="item" onChange={onChange} />
+        </Grommet>,
+      );
+      fireEvent.change(getByPlaceholderText('item'), {
+        target: { value: 'o' },
```
Comment (ShimiSun) [inline src/js/components/TextArea/__tests__/TextArea-test.js:187]: The snapshot needs to be aligned with this new value.

**Comment by JoeButler7** [comment]
### Update:
changed name of tests and removed snapshots that were not showing any changes. In `onChange` removed snapshot and instead assert the value of the textArea after change event

**Pull request merged**